instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/rules/TomcatTest.java

```diff
@@ -44,15 +40,14 @@ void testCollectedMetrics(String dockerImageName, String sampleWebApplicationUrl
             .withExposedPorts(8080)
             .waitingFor(Wait.forListeningPorts(8080));
 
-    copyFilesToTarget(target, yamlFiles);
-
-    startTarget(target);
+    copyAgentToTarget(target);
+    copyYamlFilesToTarget(target, yamlFiles);
 
     // Deploy example web application to the tomcat to enable reporting tomcat.session.active.count
     // metric
-    target.execInContainer("rm", "-fr", "/usr/local/tomcat/webapps/ROOT");
-    target.execInContainer(
-        "curl", sampleWebApplicationUrl, "-o", "/usr/local/tomcat/webapps/ROOT.war");
+    copyTestWebAppToTarget(target, "/usr/local/tomcat/webapps/ROOT.war");
```

Comment:
[for reviewer] we use a locally built app instead of downloading one from a remote URL, that removes dependency on network availability and also makes test likely more reliable.